### DISCUSSION TITLE
Image field resolves disk twice in URL in CRUD in 6.6.3

### DISCUSSION THREAD

**Discussion by lieto-dk** [Laravel-Backpack/community-forum#841] [Q&A (Help)] [answered]
Hi,

I have updated to Backpack Crud 6.6.3 and pro 2.1.3 and my image fields in the CRUDs now resolves the disk twice.

i have this field in my crud

        $this->crud->addField([
            'name' => "image",
            'type' => 'image',
            'label' => "Image",
            'disk' => "objectstorage",
            'crop' => true, // set to true to allow cropping, false to disable
            'aspect_ratio' => 0,525, // omit or set to 0 to allow any aspect ratio
            'wrapper'   => ['class' => 'form-group col-md-12'],
            'tab'             => 'Content (Image)',
        ]);

No accessors in the Model.

The image field is resolved to

https://storage.lastminutecharter.eu/storage.lastminutecharter.eu/media/countries/eaf982d89cf472b6b9d85b2fe6b4eb61.webp

the image field in the db has this value

/media/countries/eaf982d89cf472b6b9d85b2fe6b4eb61.webp

The storage is defined in config/filesýstems

        'objectstorage' => [
            'driver' => 's3',
            'key' => env('LINODE_ACCESS_KEY_ID'),
            'secret' => env('LINODE_SECRET_ACCESS_KEY'),
            'region' => env('LINODE_DEFAULT_REGION'),
            'bucket' => env('LINODE_BUCKET'),
            'url' => env('LINODE_URL'),
            'endpoint' => env('LINODE_ENDPOINT'),
            'use_path_style_endpoint' => env('LINODE_PATH_STYLE', false),
        ],

and in env.

LINODE_ACCESS_KEY_ID=xxxxxxxx
LINODE_SECRET_ACCESS_KEY=xxxxxx
LINODE_DEFAULT_REGION=eu-central-1
LINODE_BUCKET=storage.lastminutecharter.eu
LINODE_URL=https://storage.lastminutecharter.eu/
LINODE_ENDPOINT=https://eu-central-1.linodeobjects.com
LINODE_PATH_STYLE=true

Any ideas on what is wrong here. It worked in previous version of backpack crud (6.5.2) and pro (2.0.23).

**Comment by pxpm** [comment]
I've did changes here last few days. Sorry I am the one to blame for breaking it 😞 

I've some question that may help me to debug this issue:
- is your bucket really called `storage.lastminutecharter.eu` ? We weren't using the bucket name in previous versions, but we are now so the issue may lie there. 
- does it work if you add `temporary => true` in your image field definition ? Alternatively if you set `config.filesystems.cloud => 'objectstorage'` in the `config/filesystems.php` file ?

Cheers

**Reply by lieto-dk** [reply]
Yes, the buckets name is storage.lastminutecharter.eu. If it wasnt the URL it would not be possible in Akamai/Linodes Object Storage implementation to add a valid SSL certificate to the bucket !
S3 compatible implementations like Akamai/Linode, Idrive, Digital Ocean and soon Hetzners have a lot of small quirks so i think you should go for the simplest implementation (just using the url parameter as prefix) and if your customers wants anything more fancy they can implement it in there models or cruds !

Adding 'temporary' => true fixed the issue. I tested both viewing the image in the crud and uploading a new. What does the parameter do ?

I added  'config.filesystems.cloud' => 'objectstorage', both before and after the definition of objectstorage it didnt fix it.

I need to know whether i should add 'temporary' => true to all image fields at this time. What is the timeframe for a temporary solution ? If i add it will i then have to remove it later on ?

**Reply by lieto-dk** [reply]
I also tested if i could omit the bucket name in my env. But then the upload via the S3 Compatible filesystem driver in Laravel does not work !

**Reply by pxpm** [reply, marked as answer]
I've just released `backpack/pro v2.1.5` with the fix. 
It should take a bit for our private repository to pick up the changes and you will able to get them with a composer update. 

It was a mistake I made due to a bad configuration in my environment. It didn't fail on my environment, because it was misconfigured 🤷 

Sorry for that, and let me know if it's working as expected for you now. 

Cheers

**Reply by lieto-dk** [reply]
All god after update to pro 2.1.5. Thanks !
